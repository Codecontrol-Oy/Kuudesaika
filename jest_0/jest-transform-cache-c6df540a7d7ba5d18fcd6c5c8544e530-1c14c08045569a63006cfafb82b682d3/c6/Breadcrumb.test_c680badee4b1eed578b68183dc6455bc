'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _Breadcrumb = require('./Breadcrumb');var _Breadcrumb2 = _interopRequireDefault(_Breadcrumb);
var _Geometries = require('../../../mock/Geometries.json');var _Geometries2 = _interopRequireDefault(_Geometries);
var _reactRouter = require('react-router');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('<Breadcrumb />', function () {

    it('Should match snapshot', function () {
        var breadcrumb = shallow(
        _react2.default.createElement(_Breadcrumb2.default, { path: _reactRouter.browserHistory.getCurrentLocation() }));

        expect(breadcrumb).toMatchSnapshot();
    });

    it('Should should render base anchors', function () {
        var breadcrumb = shallow(
        _react2.default.createElement(_Breadcrumb2.default, { path: _reactRouter.browserHistory.getCurrentLocation() }));

        expect(breadcrumb.find('Anchor').length).toEqual(2);

    });
});