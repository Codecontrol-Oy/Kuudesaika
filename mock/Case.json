{
    "function": {
            "data" : {
                "url": "https://paatos-helsinki.6aika.fi/v1/function/1/",
                "id": 1,
                "data_source": "open_ahjo",
                "created_at": "2017-05-05T06:13:06.497373Z",
                "modified_at": "2017-05-05T06:13:06.497436Z",
                "origin_id": "1",
                "name": "Hallintoasiat",
                "function_id": "00",
                "parent": null               
            }
    },
    "cases": [
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/case/1/",
            "id": 1,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/20/"
            ],
            "geometries": [],
            "created_at": "2017-05-05T06:14:45.974122Z",
            "modified_at": "2017-05-05T06:14:45.974158Z",
            "origin_id": "7",
            "title": "Asuntolautakunnan esittelysuunnitelma kevätkaudelle 2012",
            "register_id": "HEL 2012-002954",
            "function": "https://paatos-helsinki.6aika.fi/v1/function/9/",
            "attachments": []
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/case/2/",
            "id": 2,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/9293/"
            ],
            "geometries": [],
            "created_at": "2017-05-05T06:14:45.986709Z",
            "modified_at": "2017-05-05T06:14:45.986746Z",
            "origin_id": "3995",
            "title": "Staran kaupunkitekniikoiden läntisen alueen pienet jyräpäällystystyöt 2013",
            "register_id": "HEL 2012-016286",
            "function": "https://paatos-helsinki.6aika.fi/v1/function/194/",
            "attachments": []
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/case/3/",
            "id": 3,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/39/"
            ],
            "geometries": [],
            "created_at": "2017-05-05T06:14:45.997440Z",
            "modified_at": "2017-05-05T06:14:45.997473Z",
            "origin_id": "16",
            "title": "Asukkaiden valinta vapautuviin ARA-asuntoihin",
            "register_id": "HEL 2012-005707",
            "function": "https://paatos-helsinki.6aika.fi/v1/function/450/",
            "attachments": []
        }]
}