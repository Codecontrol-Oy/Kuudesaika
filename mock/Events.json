{
    "events": [
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/1/",
            "id": 1,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/622/",
                "https://paatos-helsinki.6aika.fi/v1/action/623/",
                "https://paatos-helsinki.6aika.fi/v1/action/624/",
                "https://paatos-helsinki.6aika.fi/v1/action/625/"
            ],
            "created_at": "2017-05-05T06:13:12.129687Z",
            "modified_at": "2017-05-05T06:13:12.129738Z",
            "origin_id": "60",
            "name": "",
            "start_date": "2013-04-18",
            "end_date": "2013-04-18",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/219/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/2/",
            "id": 2,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/44/",
                "https://paatos-helsinki.6aika.fi/v1/action/45/",
                "https://paatos-helsinki.6aika.fi/v1/action/46/",
                "https://paatos-helsinki.6aika.fi/v1/action/47/",
                "https://paatos-helsinki.6aika.fi/v1/action/48/",
                "https://paatos-helsinki.6aika.fi/v1/action/49/"
            ],
            "created_at": "2017-05-05T06:13:12.145703Z",
            "modified_at": "2017-05-05T06:13:12.145745Z",
            "origin_id": "6",
            "name": "",
            "start_date": "2012-04-26",
            "end_date": "2012-04-26",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/3/",
            "id": 3,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/50/",
                "https://paatos-helsinki.6aika.fi/v1/action/51/",
                "https://paatos-helsinki.6aika.fi/v1/action/52/",
                "https://paatos-helsinki.6aika.fi/v1/action/53/",
                "https://paatos-helsinki.6aika.fi/v1/action/54/",
                "https://paatos-helsinki.6aika.fi/v1/action/55/",
                "https://paatos-helsinki.6aika.fi/v1/action/56/",
                "https://paatos-helsinki.6aika.fi/v1/action/57/",
                "https://paatos-helsinki.6aika.fi/v1/action/58/",
                "https://paatos-helsinki.6aika.fi/v1/action/59/",
                "https://paatos-helsinki.6aika.fi/v1/action/60/",
                "https://paatos-helsinki.6aika.fi/v1/action/61/"
            ],
            "created_at": "2017-05-05T06:13:12.157868Z",
            "modified_at": "2017-05-05T06:13:12.157908Z",
            "origin_id": "7",
            "name": "",
            "start_date": "2012-05-24",
            "end_date": "2012-05-24",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/4/",
            "id": 4,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/62/",
                "https://paatos-helsinki.6aika.fi/v1/action/63/",
                "https://paatos-helsinki.6aika.fi/v1/action/64/",
                "https://paatos-helsinki.6aika.fi/v1/action/65/",
                "https://paatos-helsinki.6aika.fi/v1/action/66/",
                "https://paatos-helsinki.6aika.fi/v1/action/67/",
                "https://paatos-helsinki.6aika.fi/v1/action/68/",
                "https://paatos-helsinki.6aika.fi/v1/action/69/",
                "https://paatos-helsinki.6aika.fi/v1/action/70/",
                "https://paatos-helsinki.6aika.fi/v1/action/71/"
            ],
            "created_at": "2017-05-05T06:13:12.169167Z",
            "modified_at": "2017-05-05T06:13:12.169206Z",
            "origin_id": "8",
            "name": "",
            "start_date": "2012-06-07",
            "end_date": "2012-06-07",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/5/",
            "id": 5,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/72/",
                "https://paatos-helsinki.6aika.fi/v1/action/73/",
                "https://paatos-helsinki.6aika.fi/v1/action/74/",
                "https://paatos-helsinki.6aika.fi/v1/action/75/",
                "https://paatos-helsinki.6aika.fi/v1/action/76/",
                "https://paatos-helsinki.6aika.fi/v1/action/77/",
                "https://paatos-helsinki.6aika.fi/v1/action/78/",
                "https://paatos-helsinki.6aika.fi/v1/action/79/",
                "https://paatos-helsinki.6aika.fi/v1/action/80/"
            ],
            "created_at": "2017-05-05T06:13:12.181149Z",
            "modified_at": "2017-05-05T06:13:12.181191Z",
            "origin_id": "9",
            "name": "",
            "start_date": "2012-08-16",
            "end_date": "2012-08-16",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/6/",
            "id": 6,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/81/",
                "https://paatos-helsinki.6aika.fi/v1/action/82/",
                "https://paatos-helsinki.6aika.fi/v1/action/83/",
                "https://paatos-helsinki.6aika.fi/v1/action/84/",
                "https://paatos-helsinki.6aika.fi/v1/action/85/",
                "https://paatos-helsinki.6aika.fi/v1/action/86/",
                "https://paatos-helsinki.6aika.fi/v1/action/87/",
                "https://paatos-helsinki.6aika.fi/v1/action/88/"
            ],
            "created_at": "2017-05-05T06:13:12.197292Z",
            "modified_at": "2017-05-05T06:13:12.197354Z",
            "origin_id": "10",
            "name": "",
            "start_date": "2012-09-27",
            "end_date": "2012-09-27",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/7/",
            "id": 7,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/89/",
                "https://paatos-helsinki.6aika.fi/v1/action/90/",
                "https://paatos-helsinki.6aika.fi/v1/action/91/",
                "https://paatos-helsinki.6aika.fi/v1/action/92/",
                "https://paatos-helsinki.6aika.fi/v1/action/93/",
                "https://paatos-helsinki.6aika.fi/v1/action/94/",
                "https://paatos-helsinki.6aika.fi/v1/action/95/",
                "https://paatos-helsinki.6aika.fi/v1/action/96/",
                "https://paatos-helsinki.6aika.fi/v1/action/97/",
                "https://paatos-helsinki.6aika.fi/v1/action/98/",
                "https://paatos-helsinki.6aika.fi/v1/action/99/",
                "https://paatos-helsinki.6aika.fi/v1/action/100/",
                "https://paatos-helsinki.6aika.fi/v1/action/101/"
            ],
            "created_at": "2017-05-05T06:13:12.210275Z",
            "modified_at": "2017-05-05T06:13:12.210315Z",
            "origin_id": "11",
            "name": "",
            "start_date": "2012-10-11",
            "end_date": "2012-10-11",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/8/",
            "id": 8,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/102/",
                "https://paatos-helsinki.6aika.fi/v1/action/103/",
                "https://paatos-helsinki.6aika.fi/v1/action/104/",
                "https://paatos-helsinki.6aika.fi/v1/action/105/",
                "https://paatos-helsinki.6aika.fi/v1/action/106/",
                "https://paatos-helsinki.6aika.fi/v1/action/107/",
                "https://paatos-helsinki.6aika.fi/v1/action/108/",
                "https://paatos-helsinki.6aika.fi/v1/action/109/",
                "https://paatos-helsinki.6aika.fi/v1/action/110/"
            ],
            "created_at": "2017-05-05T06:13:12.222358Z",
            "modified_at": "2017-05-05T06:13:12.222398Z",
            "origin_id": "12",
            "name": "",
            "start_date": "2012-10-25",
            "end_date": "2012-10-25",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/9/",
            "id": 9,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/111/",
                "https://paatos-helsinki.6aika.fi/v1/action/112/",
                "https://paatos-helsinki.6aika.fi/v1/action/113/",
                "https://paatos-helsinki.6aika.fi/v1/action/114/",
                "https://paatos-helsinki.6aika.fi/v1/action/115/",
                "https://paatos-helsinki.6aika.fi/v1/action/116/",
                "https://paatos-helsinki.6aika.fi/v1/action/117/",
                "https://paatos-helsinki.6aika.fi/v1/action/118/",
                "https://paatos-helsinki.6aika.fi/v1/action/119/"
            ],
            "created_at": "2017-05-05T06:13:12.235094Z",
            "modified_at": "2017-05-05T06:13:12.235148Z",
            "origin_id": "13",
            "name": "",
            "start_date": "2012-11-08",
            "end_date": "2012-11-08",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/10/",
            "id": 10,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/120/",
                "https://paatos-helsinki.6aika.fi/v1/action/121/",
                "https://paatos-helsinki.6aika.fi/v1/action/122/",
                "https://paatos-helsinki.6aika.fi/v1/action/123/",
                "https://paatos-helsinki.6aika.fi/v1/action/124/",
                "https://paatos-helsinki.6aika.fi/v1/action/125/",
                "https://paatos-helsinki.6aika.fi/v1/action/126/",
                "https://paatos-helsinki.6aika.fi/v1/action/127/",
                "https://paatos-helsinki.6aika.fi/v1/action/128/",
                "https://paatos-helsinki.6aika.fi/v1/action/129/",
                "https://paatos-helsinki.6aika.fi/v1/action/130/"
            ],
            "created_at": "2017-05-05T06:13:12.248815Z",
            "modified_at": "2017-05-05T06:13:12.248857Z",
            "origin_id": "14",
            "name": "",
            "start_date": "2012-11-22",
            "end_date": "2012-11-22",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/11/",
            "id": 11,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/131/",
                "https://paatos-helsinki.6aika.fi/v1/action/132/",
                "https://paatos-helsinki.6aika.fi/v1/action/133/",
                "https://paatos-helsinki.6aika.fi/v1/action/134/",
                "https://paatos-helsinki.6aika.fi/v1/action/135/",
                "https://paatos-helsinki.6aika.fi/v1/action/136/",
                "https://paatos-helsinki.6aika.fi/v1/action/137/",
                "https://paatos-helsinki.6aika.fi/v1/action/138/",
                "https://paatos-helsinki.6aika.fi/v1/action/139/",
                "https://paatos-helsinki.6aika.fi/v1/action/140/"
            ],
            "created_at": "2017-05-05T06:13:12.260504Z",
            "modified_at": "2017-05-05T06:13:12.260548Z",
            "origin_id": "15",
            "name": "",
            "start_date": "2012-12-20",
            "end_date": "2012-12-20",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/12/",
            "id": 12,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/141/",
                "https://paatos-helsinki.6aika.fi/v1/action/142/",
                "https://paatos-helsinki.6aika.fi/v1/action/143/",
                "https://paatos-helsinki.6aika.fi/v1/action/144/",
                "https://paatos-helsinki.6aika.fi/v1/action/145/",
                "https://paatos-helsinki.6aika.fi/v1/action/146/",
                "https://paatos-helsinki.6aika.fi/v1/action/147/",
                "https://paatos-helsinki.6aika.fi/v1/action/148/",
                "https://paatos-helsinki.6aika.fi/v1/action/149/",
                "https://paatos-helsinki.6aika.fi/v1/action/150/",
                "https://paatos-helsinki.6aika.fi/v1/action/151/",
                "https://paatos-helsinki.6aika.fi/v1/action/152/",
                "https://paatos-helsinki.6aika.fi/v1/action/153/",
                "https://paatos-helsinki.6aika.fi/v1/action/154/"
            ],
            "created_at": "2017-05-05T06:13:12.274474Z",
            "modified_at": "2017-05-05T06:13:12.274549Z",
            "origin_id": "16",
            "name": "",
            "start_date": "2013-02-05",
            "end_date": "2013-02-05",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/13/",
            "id": 13,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/165/",
                "https://paatos-helsinki.6aika.fi/v1/action/166/",
                "https://paatos-helsinki.6aika.fi/v1/action/167/",
                "https://paatos-helsinki.6aika.fi/v1/action/168/",
                "https://paatos-helsinki.6aika.fi/v1/action/169/",
                "https://paatos-helsinki.6aika.fi/v1/action/170/",
                "https://paatos-helsinki.6aika.fi/v1/action/171/",
                "https://paatos-helsinki.6aika.fi/v1/action/172/",
                "https://paatos-helsinki.6aika.fi/v1/action/173/"
            ],
            "created_at": "2017-05-05T06:13:12.289181Z",
            "modified_at": "2017-05-05T06:13:12.289221Z",
            "origin_id": "18",
            "name": "",
            "start_date": "2013-03-14",
            "end_date": "2013-03-14",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/14/",
            "id": 14,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/174/",
                "https://paatos-helsinki.6aika.fi/v1/action/175/",
                "https://paatos-helsinki.6aika.fi/v1/action/176/",
                "https://paatos-helsinki.6aika.fi/v1/action/177/",
                "https://paatos-helsinki.6aika.fi/v1/action/178/",
                "https://paatos-helsinki.6aika.fi/v1/action/179/"
            ],
            "created_at": "2017-05-05T06:13:12.300764Z",
            "modified_at": "2017-05-05T06:13:12.300802Z",
            "origin_id": "19",
            "name": "",
            "start_date": "2013-04-03",
            "end_date": "2013-04-03",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/15/",
            "id": 15,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/180/",
                "https://paatos-helsinki.6aika.fi/v1/action/181/",
                "https://paatos-helsinki.6aika.fi/v1/action/182/",
                "https://paatos-helsinki.6aika.fi/v1/action/183/",
                "https://paatos-helsinki.6aika.fi/v1/action/184/",
                "https://paatos-helsinki.6aika.fi/v1/action/185/",
                "https://paatos-helsinki.6aika.fi/v1/action/186/",
                "https://paatos-helsinki.6aika.fi/v1/action/187/",
                "https://paatos-helsinki.6aika.fi/v1/action/188/"
            ],
            "created_at": "2017-05-05T06:13:12.314224Z",
            "modified_at": "2017-05-05T06:13:12.314296Z",
            "origin_id": "20",
            "name": "",
            "start_date": "2013-04-25",
            "end_date": "2013-04-25",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/16/",
            "id": 16,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/196/",
                "https://paatos-helsinki.6aika.fi/v1/action/197/",
                "https://paatos-helsinki.6aika.fi/v1/action/189/",
                "https://paatos-helsinki.6aika.fi/v1/action/190/",
                "https://paatos-helsinki.6aika.fi/v1/action/191/",
                "https://paatos-helsinki.6aika.fi/v1/action/192/",
                "https://paatos-helsinki.6aika.fi/v1/action/193/",
                "https://paatos-helsinki.6aika.fi/v1/action/194/",
                "https://paatos-helsinki.6aika.fi/v1/action/195/",
                "https://paatos-helsinki.6aika.fi/v1/action/198/",
                "https://paatos-helsinki.6aika.fi/v1/action/199/"
            ],
            "created_at": "2017-05-05T06:13:12.327751Z",
            "modified_at": "2017-05-05T06:13:12.327806Z",
            "origin_id": "21",
            "name": "",
            "start_date": "2013-05-23",
            "end_date": "2013-05-23",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/17/",
            "id": 17,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/200/",
                "https://paatos-helsinki.6aika.fi/v1/action/201/",
                "https://paatos-helsinki.6aika.fi/v1/action/202/",
                "https://paatos-helsinki.6aika.fi/v1/action/203/",
                "https://paatos-helsinki.6aika.fi/v1/action/204/",
                "https://paatos-helsinki.6aika.fi/v1/action/205/",
                "https://paatos-helsinki.6aika.fi/v1/action/206/",
                "https://paatos-helsinki.6aika.fi/v1/action/207/",
                "https://paatos-helsinki.6aika.fi/v1/action/208/",
                "https://paatos-helsinki.6aika.fi/v1/action/209/"
            ],
            "created_at": "2017-05-05T06:13:12.341885Z",
            "modified_at": "2017-05-05T06:13:12.341943Z",
            "origin_id": "22",
            "name": "",
            "start_date": "2013-06-12",
            "end_date": "2013-06-12",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/157/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/18/",
            "id": 18,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/210/",
                "https://paatos-helsinki.6aika.fi/v1/action/211/",
                "https://paatos-helsinki.6aika.fi/v1/action/212/",
                "https://paatos-helsinki.6aika.fi/v1/action/213/",
                "https://paatos-helsinki.6aika.fi/v1/action/214/",
                "https://paatos-helsinki.6aika.fi/v1/action/215/",
                "https://paatos-helsinki.6aika.fi/v1/action/216/",
                "https://paatos-helsinki.6aika.fi/v1/action/217/",
                "https://paatos-helsinki.6aika.fi/v1/action/218/",
                "https://paatos-helsinki.6aika.fi/v1/action/219/",
                "https://paatos-helsinki.6aika.fi/v1/action/220/",
                "https://paatos-helsinki.6aika.fi/v1/action/221/",
                "https://paatos-helsinki.6aika.fi/v1/action/222/"
            ],
            "created_at": "2017-05-05T06:13:12.355510Z",
            "modified_at": "2017-05-05T06:13:12.355559Z",
            "origin_id": "23",
            "name": "",
            "start_date": "2012-01-25",
            "end_date": "2012-01-25",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/156/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/19/",
            "id": 19,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/223/",
                "https://paatos-helsinki.6aika.fi/v1/action/224/",
                "https://paatos-helsinki.6aika.fi/v1/action/225/",
                "https://paatos-helsinki.6aika.fi/v1/action/226/",
                "https://paatos-helsinki.6aika.fi/v1/action/227/",
                "https://paatos-helsinki.6aika.fi/v1/action/228/",
                "https://paatos-helsinki.6aika.fi/v1/action/229/",
                "https://paatos-helsinki.6aika.fi/v1/action/230/",
                "https://paatos-helsinki.6aika.fi/v1/action/231/",
                "https://paatos-helsinki.6aika.fi/v1/action/232/",
                "https://paatos-helsinki.6aika.fi/v1/action/233/",
                "https://paatos-helsinki.6aika.fi/v1/action/234/",
                "https://paatos-helsinki.6aika.fi/v1/action/235/",
                "https://paatos-helsinki.6aika.fi/v1/action/236/"
            ],
            "created_at": "2017-05-05T06:13:12.368546Z",
            "modified_at": "2017-05-05T06:13:12.368588Z",
            "origin_id": "24",
            "name": "",
            "start_date": "2012-02-08",
            "end_date": "2012-02-08",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/156/"
        },
        {
            "url": "https://paatos-helsinki.6aika.fi/v1/event/20/",
            "id": 20,
            "data_source": "open_ahjo",
            "actions": [
                "https://paatos-helsinki.6aika.fi/v1/action/237/",
                "https://paatos-helsinki.6aika.fi/v1/action/238/",
                "https://paatos-helsinki.6aika.fi/v1/action/239/",
                "https://paatos-helsinki.6aika.fi/v1/action/240/",
                "https://paatos-helsinki.6aika.fi/v1/action/241/",
                "https://paatos-helsinki.6aika.fi/v1/action/242/",
                "https://paatos-helsinki.6aika.fi/v1/action/243/",
                "https://paatos-helsinki.6aika.fi/v1/action/244/",
                "https://paatos-helsinki.6aika.fi/v1/action/245/",
                "https://paatos-helsinki.6aika.fi/v1/action/246/",
                "https://paatos-helsinki.6aika.fi/v1/action/247/",
                "https://paatos-helsinki.6aika.fi/v1/action/248/",
                "https://paatos-helsinki.6aika.fi/v1/action/249/",
                "https://paatos-helsinki.6aika.fi/v1/action/250/",
                "https://paatos-helsinki.6aika.fi/v1/action/251/"
            ],
            "created_at": "2017-05-05T06:13:12.379049Z",
            "modified_at": "2017-05-05T06:13:12.379086Z",
            "origin_id": "25",
            "name": "",
            "start_date": "2012-03-07",
            "end_date": "2012-03-07",
            "organization": "https://paatos-helsinki.6aika.fi/v1/organization/156/"
        }
    ]
}