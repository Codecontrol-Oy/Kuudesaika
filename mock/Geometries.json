{
     "geometries": [
        {
            "data_source": "open_ahjo",
            "created_at": "2017-05-05T06:13:41.855615Z",
            "modified_at": "2017-05-05T06:13:41.855672Z",
            "origin_id": "5",
            "name": "Alakiventie 5",
            "type": "address",
            "geometry": {
                "type": "Point",
                "coordinates": [
                    25.077758683528398,
                    60.22071379562326
                ]
            }
        },
        {
            "data_source": "open_ahjo",
            "created_at": "2017-05-05T06:13:47.412236Z",
            "modified_at": "2017-05-05T06:13:47.412271Z",
            "origin_id": "827",
            "name": "91-45-145-5",
            "type": "plan_unit",
            "geometry": {
                "type": "Polygon",
                "coordinates": [
                    [
                        [
                            25.078492370173475,
                            60.22155576522049
                        ],
                        [
                            25.07898612923674,
                            60.2209582329813
                        ],
                        [
                            25.07899939996638,
                            60.2209399601969
                        ],
                        [
                            25.07900944001587,
                            60.22092117773851
                        ],
                        [
                            25.079016186612012,
                            60.220902038225596
                        ],
                        [
                            25.079019567918284,
                            60.22088267633203
                        ],
                        [
                            25.079019575286875,
                            60.22086324464457
                        ],
                        [
                            25.079016200048645,
                            60.22084388677433
                        ],
                        [
                            25.079009478637996,
                            60.22082474630537
                        ],
                        [
                            25.07899944748894,
                            60.22080596682146
                        ],
                        [
                            25.07898617909662,
                            60.22078768290929
                        ],
                        [
                            25.07896979110206,
                            60.22077004707923
                        ],
                        [
                            25.07895040105921,
                            60.22075317594003
                        ],
                        [
                            25.07892813560655,
                            60.22073721302105
                        ],
                        [
                            25.07779082358496,
                            60.219992264434524
                        ],
                        [
                            25.07779082358496,
                            60.219992264434524
                        ],
                        [
                            25.07778102238484,
                            60.219986382334596
                        ],
                        [
                            25.077770247993417,
                            60.21998093162229
                        ],
                        [
                            25.077758581722932,
                            60.21997596610199
                        ],
                        [
                            25.077746104800237,
                            60.219971503676604
                        ],
                        [
                            25.077732907536436,
                            60.21996758916984
                        ],
                        [
                            25.077719098219642,
                            60.21996424046887
                        ],
                        [
                            25.077704758118717,
                            60.21996149342736
                        ],
                        [
                            25.077690004542095,
                            60.2199593659272
                        ],
                        [
                            25.077674945756275,
                            60.21995786688031
                        ],
                        [
                            25.077659681028432,
                            60.219957014179165
                        ],
                        [
                            25.07680422347679,
                            60.21992742663691
                        ],
                        [
                            25.076651548121873,
                            60.22101977300276
                        ],
                        [
                            25.07779196202046,
                            60.22153154925718
                        ],
                        [
                            25.078492370173475,
                            60.22155576522049
                        ]
                    ]
                ]
            }
        }
    ]
}