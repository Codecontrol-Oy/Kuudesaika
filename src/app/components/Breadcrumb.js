import React from 'react';
import PropTypes from 'prop-types';
import Anchor from 'grommet/components/Anchor';
import Paragraph from 'grommet/components/Paragraph';
import {getCity} from 'Actions';

export default class Breadcrumb extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
        action: null
    };
  }

  createCrumb = () => {

    const path = this.props.path.pathname;
    const pathArray = path.split('/');
    let crumb = null;
    if(pathArray.length > 1 && pathArray.length < 3) {
        crumb = <div><Anchor href={'/'}>Kaupunkivalinta</Anchor> - {getCity()}</div>
    } else if (pathArray.length < 4){
        crumb = <div><Anchor href={'/'}>Kaupunkivalinta</Anchor> - <Anchor href={'/' + getCity()}>{getCity()}</Anchor></div>
    } else {
        crumb = <div><Anchor href={'/'}>Kaupunkivalinta</Anchor> - <Anchor href={'/' + getCity()}>{getCity()}</Anchor> - {pathArray[2]}</div>
    }
    return crumb;
  }

  render () {
    const crumb = this.createCrumb();
    return (
        <Paragraph style={{marginLeft: "12px", marginTop: "0px", marginBottom: "0px"}}>
            {crumb}
        </Paragraph>
    );
  }
}

Breadcrumb.propTypes = {
    path: PropTypes.object,
};