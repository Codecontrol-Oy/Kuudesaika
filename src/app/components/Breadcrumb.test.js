import React from 'react';
import Breadcrumb from './Breadcrumb';
import mock from '../../../mock/Geometries.json';
import {browserHistory} from 'react-router';

describe('<Breadcrumb />', () => {

    it('Should match snapshot', () => {
        const breadcrumb = shallow(
            <Breadcrumb path={browserHistory.getCurrentLocation()} />
        );
        expect(breadcrumb).toMatchSnapshot();
    });

    it('Should should render base anchors', () => {
        const breadcrumb = shallow(
            <Breadcrumb path={browserHistory.getCurrentLocation()} />
        );
        expect(breadcrumb.find('Anchor').length).toEqual(2);

    });
});