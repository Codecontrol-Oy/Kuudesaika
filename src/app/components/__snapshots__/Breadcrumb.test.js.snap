// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> Should match snapshot 1`] = `
<Paragraph
  style={
    Object {
      "marginBottom": "0px",
      "marginLeft": "12px",
      "marginTop": "0px",
    }
  }
>
  <div>
    <Anchor
      animateIcon={true}
      href="/"
      method="push"
      tag="a"
    >
      Kaupunkivalinta
    </Anchor>
     - 
    <Anchor
      animateIcon={true}
      href="/null"
      method="push"
      tag="a"
    />
  </div>
</Paragraph>
`;
